### PR TITLE
Psoc6 add protection

### PR DESCRIPTION
**Pull request by OlegKapshii**

### CODE REVIEW

**Comment by OlegKapshii** [comment]
I see a number of fails, but cannot find where to see details - what is wrong. How to find the problem roots?

**Comment by dbkinder** [comment]
@nashif The shippable reports are now showing for the failures...

File: samples/security/protection_psoc6/README.rst
Comment (dbkinder) [inline samples/security/protection_psoc6/README.rst]: the continuation of the numbered list item must be indented to the text in the previous line:
```suggestion
   flash the board:
```

File: samples/security/protection_psoc6/README.rst
Comment (dbkinder) [inline samples/security/protection_psoc6/README.rst]:
```suggestion
   etc.) and connect to the board with the following settings:
```

File: samples/security/protection_psoc6/README.rst
Comment (dbkinder) [inline samples/security/protection_psoc6/README.rst]: Indent all these by three spaces too (so they're part of the numbered list item)

File: samples/security/protection_psoc6/README.rst
Comment (dbkinder) [inline samples/security/protection_psoc6/README.rst]:
```suggestion
   serial port:
```

File: samples/security/protection_psoc6/README.rst
Comment (dbkinder) [inline samples/security/protection_psoc6/README.rst]: Because you're creating the first sample in the ``samples/security`` folder, you'll also need to add another organizational file as found in the other sample folders.  Add a file ``samples/security/security.rst`` with the contents:
```
.. _security-samples:

Security Samples
################

.. toctree::
   :maxdepth: 1
   :glob:

   **/*
```

And edit ``samples/samples.rst`` to reference this new file by adding a line at the end of the toctree list there:
```
   security/security.rst
```

**Review by dbkinder — CHANGES_REQUESTED** [review]
Thanks for documenting this new sample.

**Comment by nashif** [comment]
> I see a number of fails, but cannot find where to see details - what is wrong. How to find the problem roots?

you just hit a bug, looks like the errors/warnings were too big and the comment was rejected by github and the script failed on that. Will fix that ASAP.

**Comment by OlegKapshii** [comment]
recheck

**Comment by OlegKapshii** [comment]
recheck

**Comment by OlegKapshii** [comment]
I have added more sophisticated protection configuration, that can be used for creating secure system on CM0.
But there are some problems with tests in the shippable. I cannot understand what should I do there - there no last sanity test reported. What should I do here?

**Review by dbkinder — CHANGES_REQUESTED** [review]
doc changes are almost OK

**Comment by OlegKapshii** [comment]
It seems something is wrong, but I do not understand where. It looks like the tests are not started

**Review by dbkinder — APPROVED** [review]
doc changes LGTM, thanks.

**Comment by carlescufi** [comment]
@OlegKapshii can you rebase and fix the CI issues?

**Comment by OlegKapshii** [comment]
@carlescufi  I have rebased, but there is the same problem in the Shippable tests - "Couldn't find last sanity run."
I do not know where the problem is and what I can fix to avoid this error.

**Comment by nashif** [comment]
> I do not know where the problem is and what I can fix to avoid this error.

looks like this is happening because you have nested samples. `samples/security/protection_psoc6/` has a sample.yaml then you have sample.yaml in both secure/unsecure. Which is the primary sample? you should only have one sample.yaml I guess.

**Comment by OlegKapshii** [comment]
@nashif Thank you very much. Your advice has helped - all tests passed.

**Comment by OlegKapshii** [comment]
Rebased the PR again and fixed conflicts with changes in documentation.
Can it be already merged?

**Review by agross-oss — APPROVED** [review]

File: samples/security/protection_psoc6/secure/src/main.c
Comment (agross-linaro) [inline samples/security/protection_psoc6/secure/src/main.c]: peripheral

Comment (agross-linaro) [reply samples/security/protection_psoc6/secure/src/main.c]: search and replace periferal w/ peripheral

File: samples/security/protection_psoc6/unsecure/src/main.c
Comment (agross-linaro) [inline samples/security/protection_psoc6/unsecure/src/main.c]: long term it might be good to have this as a memory region defined in DT.  This would make it less hard coded in the source and easily override-able via overlays.

**Review by agross-oss — CHANGES_REQUESTED** [review]
minor comments

File: samples/security/protection_psoc6/unsecure/src/main.c
Comment (OlegKapshii) [reply samples/security/protection_psoc6/unsecure/src/main.c]: In this case, the defines are added exclusively to this example only. The two addresses needed: one from secure and another from non-secure memory regions, to show that the CM0 core can read from any memory, but CM4 - only from non-secure.

In general, it is needed to define at least one more sram region that will be used by the secure and non-secure world for sharing data (As it is done in ARM Mbed SPM). And both worlds should know this region addresses. But, at the moment it is not critical, and, actually, I do not know how to do this in Zephyr structure.
If needed - I can dig in the code and add several more special memory regions for normal SPM functioning.

File: samples/security/protection_psoc6/secure/src/main.c
Comment (OlegKapshii) [reply samples/security/protection_psoc6/secure/src/main.c]: done

**Comment by OlegKapshii** [comment]
And what about the Codeowners test - who and for which folders should be added in the CODEOWNERS file?

**Comment by OlegKapshii** [comment]
So, any comments about the CODEOWNERS file?

**Comment by OlegKapshii** [comment]
@galak @agross-korg Should I simply add anybody from samples codeowners, or add my name in the CODEOWNERS file for /samples/security to avoid the check fail?

**Comment by agross-linaro** [comment]
@OlegKapshii Add your name as you are the original author.  If that needs to change in the future, it can.

**Comment by OlegKapshii** [comment]
So, all tests passed. @galak @agross-korg can we close this PR?

**Comment by samfriedman93** [comment]
@OlegKapshii @agross-oss @galak Is this waiting on anything? Can we update/merge this?

**Comment by OlegKapshii** [comment]
It waited for changes review and the PR merging, but there were nobody free who was able to do this. So the PR just left in the freeze state.
If anybody who can do the review is available, I can find time to solve the conflicts and push changes again.

**Pull request closed without merging**
